torchsynth/module.py

```diff
@@ -541,23 +540,43 @@ def _forward(self, audio_in: Signal, control_in: Signal) -> Signal:
 
 class Noise(SynthModule):
     """
-    Generates white noise that is the same length as the buffer
+    Generates white noise that is the same length as the buffer.
+
+    A batch size number of noise samples are pre-computed for performance.
+    A seed for the noise generator is required. If you use multiple of
+    these in a synth then choose different seeds for each instance.
     """
 
-    def __init__(self, synthglobals: SynthGlobals, **kwargs):
+    def __init__(self, synthglobals: SynthGlobals, seed: int, **kwargs):
         super().__init__(synthglobals, **kwargs)
-        self.register_buffer(
-            "noise",
-            torch.empty((self.batch_size, self.buffer_size), device=self.device),
-        )
+
+        # Pre-compute noise
+        generator = torch.Generator(device="cpu").manual_seed(seed)
+        noise = torch.empty((DEFAULT_BATCH_SIZE, self.buffer_size), device="cpu")
+        noise.data.uniform_(-1.0, 1.0, generator=generator)
+        if self.batch_size > DEFAULT_BATCH_SIZE:
+            noise = util.batch_resize(noise, self.batch_size)
+
+        self.register_buffer("noise", noise.to(self.device))
+        self.offset = 0
 
     def _forward(self) -> Signal:
-        if self.seed is not None:
-            mt19937_gen = csprng.create_mt19937_generator(self.seed)
-            self.noise.data.uniform_(-1, 1, generator=mt19937_gen)
+        # Return noise quickly if we are returning the whole batch
+        # and don't need to offset at all.
+        if self.batch_size >= DEFAULT_BATCH_SIZE and self.offset == 0:
```

Comment:
okay this self.offset thing is a bit gnarly, but I think we should just revisit this.

I have a few ideas how to fix this later